satpy/writers/core/compute.py

```diff
@@ -0,0 +1,144 @@
+# Copyright (c) 2025 Satpy developers
+#
+# This file is part of satpy.
+#
+# satpy is free software: you can redistribute it and/or modify it under the
+# terms of the GNU General Public License as published by the Free Software
+# Foundation, either version 3 of the License, or (at your option) any later
+# version.
+#
+# satpy is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# satpy.  If not, see <http://www.gnu.org/licenses/>.
+"""Utilities for writers."""
```

Comment:
Should the file be named `utils.py` or the docstring changed? Then the `compute_writer_results()` wouldn't really fit and would need its own module.

I was going to comment on `split_results()` and `group_results_by_output_file()` that they'd fit better in `grouping.py`. Or `utils.py`.

Something to think about anyways.

Reply:
`compute_writer_results()` could also be in `base.py` :thinking:

Reply:
So I had the incorrect assumption that `split_results` was needed by users. It seems users only need either `compute_writer_results` or `group_results_by_output_file` from what I can tell. The group function isn't documented outside of its docstring so I'm not sure I fully understand the use case.

The utils docstring was left over from refactoring, it is meant to be compute.py. I don't agree with `compute_writer_results` going in `base.py` as it is user-facing and base.py is more for base operations/classes. All 3 of these functions are only used when you are concerned about computing or delaying computing of the results of writers so it makes sense to me that they go in a `compute.py` module.